File: Backend/threadly-order-and-cart-management/src/main/java/online/threadly/order_and_cart_management/dto/Product.java
package online.threadly.order_and_cart_management.dto;

import lombok.Data;

import java.util.UUID;

@Data
public class Product {
    private UUID id;

    private String name;

    private String slug;

    private String[] images;

    private Double price;

    private Integer quantity;

    private String category;

    private String subcategory;

    private String color;

    private String size;

    private Double originalPrice;

    private Double discount;

    private boolean bestSeller;

    private boolean newArrival;
}

File: Frontend/src/types/product.ts
export const Category = {
  FASHION: "FASHION",
  ELECTRONICS: "ELECTRONICS",
  BEAUTY: "BEAUTY",
  HOME_APPLIANCES: "HOME_APPLIANCES",
} as const;

export type Category = (typeof Category)[keyof typeof Category];

export interface Product {
  id: string;
  name: string;
  slug: string;
  images: string[];
  brand: string;
  description: string;
  stock: number;
  price: number;
  rating: number;
  ratingCount?: number;
  rating_count?: number;
  isFeatured?: boolean;
  is_featured?: boolean;
  createdAt?: string;

  // New detailed schema fields
  category: Category | string;
  subcategory?: string;
  color?: string;
  size?: string;
  originalPrice?: number;
  discount?: number;
  bestSeller?: boolean;
  newArrival?: boolean;
}
